transport.go

```diff
@@ -171,19 +156,19 @@ func getTransport(opts TransportOptions, enc encoding.Encoding, tracer opentraci
 			RoutingKey:      opts.RoutingKey,
 			ShardKey:        opts.ShardKey,
 			Peers:           hostPorts,
-			Encoding:        enc.String(),
+			Encoding:        resolved.enc.String(),
 			TransportOpts:   opts.TransportHeaders,
 			Tracer:          tracer,
 		}
 		return transport.NewTChannel(topts)
 	}
 
-	if protocol == "grpc" {
+	if resolved.protocol == transport.GRPC {
 		return transport.NewGRPC(transport.GRPCOptions{
 			Addresses:       getHosts(opts.Peers),
 			Tracer:          tracer,
 			Caller:          opts.CallerName,
-			Encoding:        enc.String(),
+			Encoding:        resolved.enc.String(),
```

Comment:
Note to self we can bring back test+error that were deleted in https://github.com/yarpc/yab/pull/274/files#diff-7f53f14597270e73b9fb5e7892df8ce3.

Reply:
sweet!